acceptance/analyzer_test.go

```diff
@@ -1125,16 +1128,17 @@ func testAnalyzerFunc(platformAPI string) func(t *testing.T, when spec.G, it spe
 							"docker", "run", "--rm",
 							"--network", registryNetwork,
 							"--env", "CNB_PLATFORM_API="+platformAPI,
+							"--env", "CNB_REGISTRY_AUTH="+fixtures.authRegAuthConfig,
 							"--name", containerName,
 							analyzeImage,
 							ctrPath(analyzerPath),
-							"-tag", readOnlyRegistry.RepoName("my-tag"),
-							readOnlyRegistry.RepoName("some-image"),
+							"-tag", fixtures.inaccessibleImage,
```

Comment:
I'm a bit confused how this is working. The `inaccessibleImage` doesn't have any privileges defined on it, so it should fall through to the Basic Auth - but then we provided valid credentials in `fixtures.authRegAuthConfig`, so how is it failing to ensure the access?

Reply:
Well, with the change proposed in imgUtil [PR 126](https://github.com/buildpacks/imgutil/pull/126) this image can help to understand it

![wrapping 2](https://user-images.githubusercontent.com/1181799/128925869-488daafe-c531-4f2b-bcd7-a1f69fff0c20.png)

- The request to the `inaccessibleImage` fails with 401 Unauthorized, because the first filter to handle any registry request is the `Custom` and it fills the conditions (it is on permissions map), it never goes into the `Basic Auth` filter, that's why the validation fails correctly.

Reply:
> it is on permissions map

This helps answer my question [here](https://github.com/buildpacks/lifecycle/pull/685/files#r685552562). Thinking about how this could be made more clear...